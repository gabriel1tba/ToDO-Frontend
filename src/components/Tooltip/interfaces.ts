export interface ITooltip {
  title: string;
  className?: string;
  children?: React.ReactNode;
}
