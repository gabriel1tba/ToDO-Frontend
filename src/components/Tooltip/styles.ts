import styled from 'styled-components';

export const Wrapper = styled.div`
  position: relative;

  span {
    width: 160px;
    background: ${({ theme }) => theme.colors.orange.main};
    padding: 8px;
    border-radius: 4px;
    font-size: 14px;
    font-weight: 500;
    opacity: 0;
    visibility: hidden;
    transition: opacity 0.4s;

    position: absolute;
    bottom: calc(100% + 6px);
    left: 50%;
    transform: translateX(-50%);
    color: ${({ theme }) => theme.colors.gray.main};

    &::before {
      content: '';
      border-style: solid;
      border-color: ${({ theme }) => theme.colors.orange.main};
      border-width: 6px 6px 0 6px;
      top: 100%;
      position: absolute;
      left: 50%;
      transform: translateX(-50%);
    }
  }

  &:hover span {
    opacity: 1;
    visibility: visible;
  }
`;
