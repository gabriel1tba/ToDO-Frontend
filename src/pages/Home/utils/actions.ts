import { TTodo } from 'services/TodoService/interfaces';

export enum Types {
  GET_TODOS = 'GET_TODOS',
  CREATE_TODO = 'CREATE_TODO',
  UPDATE_TODO = 'UPDATE_TODO',
  DELETE_TODO = 'DELETE_TODO',
}

interface GetTodos {
  type: Types.GET_TODOS;
  payload: TTodo[];
}

interface CreateTodo {
  type: Types.CREATE_TODO;
  payload: TTodo;
}

interface UpdateTodo {
  type: Types.UPDATE_TODO;
  payload: TTodo;
}

interface DeleteTodo {
  type: Types.DELETE_TODO;
  payload: string;
}

export type Action = GetTodos | CreateTodo | UpdateTodo | DeleteTodo;

export const getTodosAction = (todos: TTodo[]): GetTodos => ({
  type: Types.GET_TODOS,
  payload: todos,
});

export const createTodoAction = (todo: TTodo): CreateTodo => ({
  type: Types.CREATE_TODO,
  payload: todo,
});

export const updateTodoAction = (todo: TTodo): UpdateTodo => ({
  type: Types.UPDATE_TODO,
  payload: todo,
});

export const deleteTodoAction = (id: string): DeleteTodo => ({
  type: Types.DELETE_TODO,
  payload: id,
});
