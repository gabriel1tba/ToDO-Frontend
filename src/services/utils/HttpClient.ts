import axios from 'axios';

const HttpClient = axios.create({
  baseURL: `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}`,
});

export default HttpClient;
