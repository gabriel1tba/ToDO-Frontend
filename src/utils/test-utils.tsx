import { render, RenderOptions } from '@testing-library/react';
import { TodoProvider } from 'context/todos';

interface IWrapperProvider {
  children: React.ReactNode;
}

const WrapperProvider = ({ children }: IWrapperProvider) => (
  <TodoProvider>{children}</TodoProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(<WrapperProvider>{ui}</WrapperProvider>, { ...options });

export * from '@testing-library/react';
export { customRender as render };
