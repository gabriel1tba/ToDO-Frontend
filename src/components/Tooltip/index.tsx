import * as S from './styles';

import { ITooltip } from './interfaces';

const Tooltip = ({ title, className, children }: ITooltip) => {
  return (
    <S.Wrapper className={className}>
      {!!children && children}
      <span>{title}</span>
    </S.Wrapper>
  );
};

export default Tooltip;
