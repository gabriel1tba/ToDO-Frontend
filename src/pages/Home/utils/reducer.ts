import { TTodo } from 'services/TodoService/interfaces';

import { Types, Action } from './actions';

const todosReducer = (state: TTodo[], action: Action): TTodo[] => {
  switch (action.type) {
    case Types.GET_TODOS: {
      return action.payload;
    }

    case Types.CREATE_TODO: {
      return [...state, action.payload];
    }

    case Types.UPDATE_TODO: {
      return state.map((todo) => {
        if (action.payload.id === todo.id) {
          todo = { ...action.payload, created_at: todo.created_at };
        }
        return todo;
      });
    }

    case Types.DELETE_TODO: {
      return state.filter((todo) => {
        return todo.id !== action.payload;
      });
    }

    default:
      return state;
  }
};

export default todosReducer;
