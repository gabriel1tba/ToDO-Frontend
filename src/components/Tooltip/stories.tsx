import { Story, Meta } from '@storybook/react';
import { FiAlertCircle } from 'react-icons/fi';

import Tooltip, { ITooltip } from '.';

export default {
  title: 'Tooltip',
  component: Tooltip,
  argTypes: {
    title: {
      type: 'string',
    },
    children: {
      type: 'function',
    },
  },
} as Meta;

export const Default: Story<ITooltip> = (args) => <Tooltip {...args} />;

Default.args = {
  title: 'Lorem Ipsum',
  children: <FiAlertCircle size={20} />,
};
