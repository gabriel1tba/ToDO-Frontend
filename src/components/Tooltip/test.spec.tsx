import { render, screen } from 'utils/test-utils';

import Tooltip from '.';

describe('<Tooltip />', () => {
  it('should render the Tooltip correctly', () => {
    const { container } = render(<Tooltip title="Item obrigatório" />);

    expect(container.firstChild).toHaveStyle({
      position: 'relative',
    });

    expect(screen.getByText(/item obrigatório/i)).toBeInTheDocument();

    expect(screen.getByText(/item obrigatório/i)).toHaveStyle({
      position: 'absolute',
      width: '160px',
      background: '#ff9000',
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
