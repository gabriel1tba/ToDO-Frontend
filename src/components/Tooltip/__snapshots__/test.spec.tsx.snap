// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should render the Tooltip correctly 1`] = `
.c0 {
  position: relative;
}

.c0 span {
  width: 160px;
  background: #ff9000;
  padding: 8px;
  border-radius: 4px;
  font-size: 14px;
  font-weight: 500;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: opacity 0.4s;
  transition: opacity 0.4s;
  position: absolute;
  bottom: calc(100% + 6px);
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  color: #424242;
}

.c0 span::before {
  content: '';
  border-style: solid;
  border-color: #ff9000;
  border-width: 6px 6px 0 6px;
  top: 100%;
  position: absolute;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c0:hover span {
  opacity: 1;
  visibility: visible;
}

<div
  class="c0"
>
  <span>
    Item obrigat√≥rio
  </span>
</div>
`;
